### PR TITLE
fix(slack): fix SlackService early return tests

### PR DESCRIPTION
**Pull request by cathteng**
These weren't actually testing anything, fixed the mocked logger to properly test where we early return in SlackService for notifying Slack threads of activities.

### CODE REVIEW

**Review by schew2381 — APPROVED** [review]
lgtm, typing has an issue tho

File: tests/sentry/integrations/slack/service/test_slack_service.py [hunk 1]
```diff
@@ -93,6 +92,7 @@ def setUp(self) -> None:
 
     def test_none_group(self):
         self.activity.update(group=None)
+        self.service._logger = mock.MagicMock()
```
Comment (asottile-sentry) [inline tests/sentry/integrations/slack/service/test_slack_service.py:95]: mypy was helping you here, monkeypatching a class like this leaks outside of the scope of the test

this should use a context manager so it is undone after the test finishes:

```python
with mock.patch.object(self.service, '_logger') as self.mock_logger:
    ...
```

Comment (cathteng) [reply tests/sentry/integrations/slack/service/test_slack_service.py:95]: i get the same mypy error when i add the context manager

Comment (asottile-sentry) [reply tests/sentry/integrations/slack/service/test_slack_service.py:95]: can you show the mypy error you're referring to?

**Pull request merged**